### PR TITLE
Avoid global configuration overriding all

### PR DESCRIPTION
**Pull request by rakyll**
It will be more common for the users to tweak a few
settings rather than all.

Rename SetConfig to ApplyConfig and only
override the non-zero values.

### CODE REVIEW

File: trace/config.go [hunk 1]
```diff
@@ -24,8 +24,10 @@ type Config struct {
...
 }
 
-// SetConfig sets the global tracing configuration.
-func SetConfig(cfg Config) {
+// ApplyConfig applies changes to the global tracing configuration.
+//
+// Fields not provided in the given config is going to be preserved.
+func ApplyConfig(cfg Config) {
 	if cfg.DefaultSampler == nil {
 		cfg.DefaultSampler = newDefaultSampler()
```
Comment (bogdandrutu) [inline trace/config.go]: Here you don't want to reinstall the default, you want to keep the previous value. Maybe worth having a test.

Comment (rakyll) [reply trace/config.go]: Done.

**Comment by ramonza** [comment]
Nice!

File: trace/config.go [hunk 2]
```diff
@@ -24,12 +24,14 @@ type Config struct {
 	DefaultSampler Sampler
 }
 
-// SetConfig sets the global tracing configuration.
-func SetConfig(cfg Config) {
+// ApplyConfig applies changes to the global tracing configuration.
+//
+// Fields not provided in the given config is going to be preserved.
```
Comment (odeke-em) [inline trace/config.go]: Minor typo:
s/is going to/are going to/

File: trace/config_test.go [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+
+import (
+	"reflect"
+	"testing"
+)
+
+func TestApplyZeroConfig(t *testing.T) {
+	cfg := config
+	ApplyConfig(Config{})
+	if got, want := reflect.ValueOf(config.DefaultSampler).Pointer(), reflect.ValueOf(cfg.DefaultSampler).Pointer(); got != want {
```
Comment (odeke-em) [inline trace/config_test.go]: Sorry, what is this testing for?

File: trace/config.go (hunk 2 above)
Comment (rakyll) [reply trace/config.go]: Done

**Pull request merged**